pooch/core.py

```diff
@@ -357,15 +363,17 @@ def mydownloader(url, output_file, pooch):
                     action_word[action], fname, self.get_url(fname), str(self.path)
                 )
             )
-            if downloader is None:
-                downloader = HTTPDownloader()
+
+            options = infer_protocol_options(url)
+            if downloader is None and options["protocol"] in KNOWN_DOWNLOADERS:
+                downloader = KNOWN_DOWNLOADERS[options["protocol"]]()
```

Comment:
```suggestion
                protocol = infer_protocol(url)
                if protocol not in KNOWN_DOWNLOADERS:
                    raise ValueError(
                        "Unrecognized URL protocol '{}' in '{}'. Must be one of {}."
                        .format(protocol, url, KNOWN_DOWNLOADERS.keys())
                    )
                downloader = KNOWN_DOWNLOADERS[protocol]()
```

Reply:
Need to check if the inferred protocol matches a known protocol and raise an error if it doesn't. Otherwise, the user would get an error like `None type object is not callable` which is not very informative. Since this adds a possibility for an exception to be raised, we need a test for this case (i.e., pass in an invalid protocol and check that the exception is raised).